satpy/readers/amsr2_l2_gaasp.py

```diff
@@ -0,0 +1,293 @@
+filename where SH stands for South Hemisphere and NH stands for North
+Hemisphere. These gridded products are on the EASE2 North pole and South pole
+grids. See https://nsidc.org/ease/ease-grid-projection-gt for more details.
+
+Note that since SEAICE products can be on both the northern or
+southern hemisphere or both depending on what files are provided to Satpy, this
+reader appends a `_NH` and `_SH` suffix to all variable names that are
+dynamically discovered from the provided files.
+
+"""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+from pyproj import CRS
+from pyresample.geometry import AreaDefinition
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy._compat import cached_property
+from satpy import CHUNK_SIZE
+
+logger = logging.getLogger(__name__)
+
+
+class GAASPFileHandler(BaseFileHandler):
+    """Generic file handler for GAASP output files."""
+
+    y_dims = (
+        'Number_of_Scans',
+    )
+    x_dims = (
+        'Number_of_hi_rez_FOVs',
+        'Number_of_low_rez_FOVs',
+    )
+    time_dims = (
+        'Time_Dimension',
+    )
+    is_gridded = False
+    dim_resolutions = {
+        'Number_of_hi_rez_FOVs': 5000,
+        'Number_of_low_rez_FOVs': 10000,
+    }
+
+    @cached_property
+    def nc(self):
+        """Get the xarray dataset for this file."""
+        chunks = {dim_name: CHUNK_SIZE for dim_name in
+                  self.y_dims + self.x_dims + self.time_dims}
+        nc = xr.open_dataset(self.filename,
+                             decode_cf=True,
+                             mask_and_scale=False,
+                             chunks=chunks)
+
+        if len(self.time_dims) == 1:
+            nc = nc.rename({self.time_dims[0]: 'time'})
+        return nc
+
+    @property
+    def start_time(self):
+        """Get start time of observation."""
+        try:
+            return self.filename_info['start_time']
+        except KeyError:
+            time_str = self.nc.attrs['time_coverage_start']
+            return datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    @property
+    def end_time(self):
+        """Get end time of observation."""
+        try:
+            return self.filename_info['end_time']
+        except KeyError:
+            time_str = self.nc.attrs['time_coverage_end']
+            return datetime.strptime(time_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+    @property
+    def sensor_names(self):
+        """Sensors who have data in this file."""
+        return {self.nc.attrs['instrument_name'].lower()}
+
+    @property
+    def platform_name(self):
+        """Name of the platform whose data is stored in this file."""
+        return self.nc.attrs['platform_name']
+
+    def _get_var_name_without_suffix(self, var_name):
+        var_suffix = self.filetype_info.get('var_suffix', "")
+        if var_suffix:
+            var_name = var_name[:-len(var_suffix)]
+        return var_name
```

Comment:
can it happen that the suffix doesn't match the var name? should there be a check for that?

Reply:
Not sure what you mean. The suffix is added to the file's variable name in `available_datasets`. The suffix is something like `_NH` or `_SH` for north/south hemisphere.

Reply:
so you are sure that var_name ends with var_suffix every time, right?

Reply:
Yes, because of these lines:

https://github.com/pytroll/satpy/pull/1421/files#diff-096de54ec4d4d671604ddac719e8dc7e77121742b9634bb7102da6c35f70f817R200-R204

Ugh, I can't get the link to work correctly, but this part which happens in the `available_datasets`:

```
        var_suffix = self.filetype_info.get('var_suffix', "")
        ds_info = {
            'file_type': self.filetype_info['file_type'],
            'name': var_name + var_suffix,
        }
```

Reply:
Ok, good.